tutorials/roostats/rs301_splot.py

```diff
@@ -0,0 +1,318 @@
+    ##                                 const char* wgtVarName = nullptr
+    ##                                 const char* cuts = nullptr
+    ## not even using hex(id(nullptr))
+    ## improve pythonzation...
+    ## ROOT.RooDataSet::RooDataSet(RooStringView name, ROOT.RooStringView title, ROOT.RooDataSet* data, const ROOT.RooArgSet& vars, const ROOT.RooFormulaVar&     cutVar, const char* wgtVarName = nullptr)
+    ## ROOT.RooDataSet::RooDataSet(RooStringView name, ROOT.RooStringView title, ROOT.RooDataSet* data, const ROOT.RooArgSet& vars, const char* cuts = nu    llptr, const char* wgtVarName = nullptr)
+    # """
+    cutVar0 = ROOT.RooFormulaVar()  # continues at ...
+    # for the purpose of this tutorial we don't use cuts.
+    # however, it is possible to implement them as shown below:
+    zYield = object.var("zYield")
+    qcdYield = object.var("qcdYield")
+    ral = ROOT.RooArgList(zYield, qcdYield)
+    cutVar = ROOT.RooFormulaVar("zYield < 0 and qcdYield < 0.5", "Cuts", " @0 < 0 && @1 < 0.5", ral, True)
+    # ... continues here
+    dataw_qcd = ROOT.RooDataSet(data.GetName(), data.GetTitle(), data, data.get(), cutVar0, "qcdYield_sw")
+    dataw_z = ROOT.RooDataSet(data.GetName(), data.GetTitle(), data, data.get(), cutVar0, "zYield_sw")
+
+    # this shouldn't be necessary, need to fix something with workspace
+    # do this to set parameters back to their fitted values.
+    # model.fitTo(data, Extended())
+
+    # plot invMass for data with full model and individual components overlaid
+    # cdata = TCanvas()
+    cdata.cd(1)
+    frame = invMass.frame(Title="Fit of model to discriminating variable")
+    data.plotOn(frame)
+    model.plotOn(frame, Name="FullModel")
+    model.plotOn(frame, Components=zModel, LineStyle="--", LineColor="r", Name="ZModel")
+    model.plotOn(frame, Components=qcdModel, LineStyle="--", LineColor="g", Name="QCDModel")
+
+    leg = ROOT.TLegend(0.11, 0.5, 0.5, 0.8)
+    leg.AddEntry(frame.findObject("FullModel"), "Full model", "L")
+    leg.AddEntry(frame.findObject("ZModel"), "Z model", "L")
+    leg.AddEntry(frame.findObject("QCDModel"), "QCD model", "L")
+    leg.SetBorderSize(0)
+    leg.SetFillStyle(0)
+
+    frame.Draw()
+    leg.DrawClone()
+
+    # Now use the sWeights to show isolation distribution for Z and QCD.
+    # The SPlot class can make this easier, but here we demonstrate in more
+    # detail how the sWeights are used.  The SPlot class should make this
+    # very easy and needs some more development.
+
+    # Plot isolation for Z component.
+    # Do this by plotting all events weighted by the sWeight for the Z component.
+    # The SPlot class adds a new variable that has the name of the corresponding
+    # yield + "_sw".
+    cdata.cd(2)
+
+    frame2 = isolation.frame(Title="Isolation distribution with s weights to project out Z")
+    # Since the data are weighted, we use SumW2 to compute the errors.
+    dataw_z.plotOn(frame2, DataError=ROOT.RooAbsData.SumW2)
```

Comment:
```suggestion
    dataw_z.plotOn(frame2, DataError="SumW2")
```
We have a Pythonization for this :slightly_smiling_face: